app/controllers/users_controller.rb

```diff
@@ -0,0 +1,53 @@
+class UsersController < ApplicationController
+  # enable :sessions
+
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'/users/signup'
+    end
+  end
+
+  post '/signup' do
+  if params[:username] == "" || params[:email] == "" || params[:password] == ""
```

Comment:
You are already using validations on the model so we don't need to check for this now.